app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailFragment.kt

```diff
@@ -106,455 +98,402 @@ class EventDetailFragment : Fragment(R.layout.fragment_event_detail) {
+            }
+
+            collectLatestLifecycleFlow(viewModel.isDone) { isDone ->
+                if (isDone) {
+                    addTitleAndDoneButtonLayout.taskDoneCircle
+                        .setBackgroundResource(R.drawable.task_done_circle)
+                    addTitleAndDoneButtonLayout.titleTextView.paintFlags =
+                        Paint.STRIKE_THRU_TEXT_FLAG
+                } else {
+                    addTitleAndDoneButtonLayout.taskDoneCircle
+                        .setBackgroundResource(R.drawable.ic_undone_circle)
+                    addTitleAndDoneButtonLayout.titleTextView.paintFlags = Paint.ANTI_ALIAS_FLAG
+                }
+            }
+
+            collectLatestLifecycleFlow(viewModel.title) { title ->
+                addTitleAndDoneButtonLayout.titleTextView.text = title
+            }
+
+            collectLatestLifecycleFlow(viewModel.description) { description ->
+                addDescriptionLayout.descriptionTextView.text = description
             }
-            navController.navigate(
-                EventDetailFragmentDirections
-                    .actionEventDetailFragmentToPhotoDetailFragment(
-                        viewModel.photos.value[index].toString(), index
-                    )
-            )
-        }
 
-        binding.addPhotoLayout.apply {
             collectLatestLifecycleFlow(viewModel.photos) { photoList ->
-                addPhotoLayout.isVisible = !isAttendee || photoList.isNotEmpty()
+                addPhotoLayout.addPhotoLayout.isVisible = !isAttendee || photoList.isNotEmpty()
 
                 val adapter = PhotoItemAdapter(
                     photoList,
                     onPhotoClick = { index -> openPhoto(index) },
                     onAddPhotoClick = { addPhoto() },
                     userIsAttendee = isAttendee
                 )
-                photosRecyclerView.adapter = adapter
-                photosRecyclerView.layoutManager = LinearLayoutManager(
+                addPhotoLayout.photosRecyclerView.adapter = adapter
+                addPhotoLayout.photosRecyclerView.layoutManager = LinearLayoutManager(
                     requireContext(), LinearLayoutManager.HORIZONTAL, false
                 )
 
-                addPhotoPlusSign.isVisible = photoList.isEmpty()
-                addPhotoTextView.isVisible = photoList.isEmpty()
-                photosTextView.isVisible = photoList.isNotEmpty()
-                photosRecyclerView.isVisible = photoList.isNotEmpty()
-            }
-            addPhotoTextView.setOnClickListener {
-                addPhoto()
+                addPhotoLayout.addPhotoPlusSign.isVisible = photoList.isEmpty()
+                addPhotoLayout.addPhotoTextView.isVisible = photoList.isEmpty()
+                addPhotoLayout.photosTextView.isVisible = photoList.isNotEmpty()
+                addPhotoLayout.photosRecyclerView.isVisible = photoList.isNotEmpty()
```

Comment:
Isn't the `addPhotoPlusSign` already included in your RecyclerView? Why do you need it as separate view then?

Reply:
This is the add photo button in the mockup outside the recyclerview, when there are no photos in the list